src/teehr/loading/nwm/grid_utils.py

```diff
@@ -29,11 +29,13 @@ def compute_zonal_mean(
     # Get the values and apply weights
     var_values = arr_2d[rows, cols]
     weights_df["value"] = var_values * weights_df.weight.values
```

Comment:
@samlamont I guess this is not from this PR, but this is a bit confusing (and maybe not a good idea) to assign the values back to the weights file.  Seems like this should be a new dataframe?  Or maybe at least use a different column name, `weighted_values` or something.

Reply:
Sounds good I was thinking it would help reduce the memory footprint but I agree it's confusing